contents/blog/machine-copy-paste-mcp-intro.mdx

```diff
@@ -0,0 +1,65 @@
+
+What used to take minutes can happen in seconds. Automation at its finest.
+
+So: copy-paste is the enemy of a good LLM developer experience. The job for developer tools is to make it unnecessary, and MCP exists to support this goal. Get exactly the right data in and out of an agent session, do it fast, don’t make the developer work.
+
+## Saving labor with MCP
+
+First, make it easy to install an MCP server. You can use an [automated tool](/blog/envoy-wizard-llm-agent) to inject the configuration on the developer’s behalf. Check out [this pull request](https://github.com/PostHog/wizard/pull/48) from our wizard for a look at how that works.
+
+Next, let’s understand the protocol’s features.
+
+### Resources
+
+**What they do**: resources dump information into the context. This could be an API key, a JSON body to help write parsing logic, or a list of established values that code must incorporate. An MCP server could get this stuff from your existing APIs and services.
+
+**How they save labor**: imagine implementing [feature flags](/docs/feature-flags/adding-feature-flag-code) in some new code. We *really* don’t want the model to invent new names for this stuff. But we also don’t want the developer to log into their account and start copying and pasting flag names. So instead, we could use MCP to pipe a list of flags and descriptions directly into the agent session.
+
+### Tools
+
+**What they do**: tools trigger actions on behalf of the user, based on the goals and information already in the context. This means you can automate complex workflows to make sure no steps are skipped, and no transcription errors happen. Like resources, your MCP server could wrap your APIs to deliver this power.
+
+**How they save labor**: back to the feature flags example. If you’re in a *brand new app* and there are no feature flag names already established, a tool could save the user lots of clicking and typing. Within seconds, the new configuration can be written to a project, creating, naming and describing all the new flags.
+
+### Prompts
+
+**What they do**: exactly what you expect. They prompt the model. The spookiest thing for me – still! – is that instead of “programming” LLMs, you just kinda tell them what to do. Some instructions produce better results than others, and so consistently using a tested prompt can improve every developer’s session.
+
+**How they save labor**: you can use both the agent session *and* conventional programming to decide *which* prompt to apply. Maybe Next.js 14 wants one prompt, while 15 wants another. Instead of a developer having to figure out the right phrasing, or read documentation to help structure a prompt, your MCP server can pick up the load. Prompts can also work in tandem with tools and resources, guiding the agent session in how to accomplish work.
+
```

Comment:
I feel like we are missing a tiny bit more.

- How do dev tools define resources, tools, and prompts?
   -  You make an MCP server, format everything in a specific way, and publish it for users to use
- How do end users make use of them
   - Download the MCP server and run it locally or the cloud??? Run it in your code editor??? I'm really not sure here?